setup.py

```diff
@@ -121,6 +122,121 @@
     'Topic :: Scientific/Engineering :: Visualization',
     ]
 
+
+class NoseTestCommand(TestCommand):
+    """Invoke unit tests using nose after an in-place build."""
```

Comment:
I am not entirely sure what I mean with "an in-place build" here. 

You should be able to run the tests after a `python setup.py install`.

I guess the original line of thought for this pull request has been lost upon rebasing. This will make things more clear. I started this pull request, because maptlotlib introduced a dependency on `nose` somewhat after version matplotlib 1.2. Nose is however not a normal runtime dependency, in fact it is only needed for executing and running the tests. So I would like matplotlib to follow the standard practice here of not specifying this as a runtime dependency.

Distutils / setuptools has a specific parameter for test dependencies, this pull request moves the test dependencies there. Apart from removing nose from the runtime dependencies (i.e. that are installed when you `pip install matplotlib`) this has 2 consequences: 

1. When running `python setup.py test` the test dependencies are installed as ``eggs`` into the project folder (not into the virtual env) and
2. a test command must be hooked in setup.py to execute the tests to execute the tests this way

You can still run tests with nose installed into the virtualenv as before (then no egg is put into the project directory, you just have to specifically `pip install nose mock`). And you can also execute the tests with the library function (i.e. from the python prompt).